jupyter_server/services/contents/filemanager.py

```diff
@@ -395,6 +395,10 @@ def get(self, path, content=True, type=None, format=None):
             if type == "directory":
                 raise web.HTTPError(400, "%s is not a directory" % path, reason="bad type")
             model = self._file_model(path, content=content, format=format)
+
+        # append file ID to model
+        model["id"] = self.file_id_manager.index(path)
```

Comment:
I think that's fine for now. Like you said, there's not much that we can do except lift the responsibility of building the response model up to `ContentsManager`.

Reply:
But it also means we need to document that this service only works with `FileContentsManager` (and its derivations).

Reply:
I think we can be a bit more precise. This service only works **by default** with `FileContentsManager` and `AsyncFileContentsManager`. Any CM implementation can use the File ID Manager just by adding the highlighted lines.

Reply:
I think this warrants some further discussion but am hesitant to unresolve this conversation again.
> Any CM implementation can use the File ID Manager just by adding the highlighted lines.

This comment is true based on the original implementation.  In that, there was no opinion made by the `FileIdManager` that the objects being indexed reside in a file system.  As a result, `ContentsManager` derivations that manage objects stored in other mediums (e.g., Databases, Cloud Object Storage, etc.) _can_ use that implementation - assuming they set the "id" into the `model`.  I hope we could figure out a way to address that one issue since the original implementation satisfied DB CMs just fine - just not file-system-based CMs.  

This latest implementation leaves any `ContentsManager` derivations that do not derive from `FileContentsManager` with no option but to implement their own `FileIdManager`.  This is the very point of [my comment](https://github.com/jupyterlab/jupyterlab/issues/12614#issuecomment-1183544389) a few weeks ago that, unless this service is decoupled from the `ContentsManager` (likely via the Event Bus as Vidar mentioned), then we might as well include it _in_ the `ContentsManager` since it's very likely the table schema will need to maintain information pertinent to the `ContentsManager` implementation.

Reply:
> As a result, ContentsManager derivations that manage objects stored in other mediums (e.g., Databases, Cloud Object Storage, etc.) can use that implementation - assuming they set the "id" into the model.

My main question on the design and this PR is: "does it support objects stored in other mediums". My impression is that is does not, which makes it useless in many cases. Happy to hear more on that.